From d768e4ade44db80071c674d9de81935ddc81179b Mon Sep 17 00:00:00 2001
From: Paul Zimmerman <pauldzim@gmail.com>
Date: Mon, 19 Feb 2024 12:00:32 -0700
Subject: [PATCH 3/3] Add script 'fix-dylibs.sh' for MacOS

This script fixes up the paths in the libAGAR 1.7.1 dylibs.
Run this after doing 'make install'.
---
 fix-dylibs.sh | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)
 create mode 100755 fix-dylibs.sh

diff --git a/fix-dylibs.sh b/fix-dylibs.sh
new file mode 100755
index 000000000..6ca8d0ed6
--- /dev/null
+++ b/fix-dylibs.sh
@@ -0,0 +1,23 @@
+#! /bin/bash
+
+# Use this on MacOS to fix up the paths in the libAGAR dylibs after installing
+
+sudo install_name_tool -id '/usr/local/lib/libag_core.dylib' /usr/local/lib/libag_core.8.dylib
+sudo install_name_tool -id '/usr/local/lib/libag_gui.dylib' /usr/local/lib/libag_gui.8.dylib
+sudo install_name_tool -id '/usr/local/lib/libag_math.dylib' /usr/local/lib/libag_math.8.dylib
+sudo install_name_tool -id '/usr/local/lib/libag_net.dylib' /usr/local/lib/libag_net.8.dylib
+sudo install_name_tool -id '/usr/local/lib/libag_sg.dylib' /usr/local/lib/libag_sg.8.dylib
+sudo install_name_tool -id '/usr/local/lib/libag_sk.dylib' /usr/local/lib/libag_sk.8.dylib
+sudo install_name_tool -id '/usr/local/lib/libag_vg.dylib' /usr/local/lib/libag_vg.8.dylib
+sudo install_name_tool -change 'libag_core.dylib' '/usr/local/lib/libag_core.dylib' /usr/local/lib/libag_gui.8.dylib
+sudo install_name_tool -change 'libag_core.dylib' '/usr/local/lib/libag_core.dylib' /usr/local/lib/libag_math.8.dylib
+sudo install_name_tool -change 'libag_gui.dylib' '/usr/local/lib/libag_gui.dylib' /usr/local/lib/libag_math.8.dylib
+sudo install_name_tool -change 'libag_core.dylib' '/usr/local/lib/libag_core.dylib' /usr/local/lib/libag_net.8.dylib
+sudo install_name_tool -change 'libag_core.dylib' '/usr/local/lib/libag_core.dylib' /usr/local/lib/libag_sg.8.dylib
+sudo install_name_tool -change 'libag_gui.dylib' '/usr/local/lib/libag_gui.dylib' /usr/local/lib/libag_sg.8.dylib
+sudo install_name_tool -change 'libag_math.dylib' '/usr/local/lib/libag_math.dylib' /usr/local/lib/libag_sg.8.dylib
+sudo install_name_tool -change 'libag_math.dylib' '/usr/local/lib/libag_math.dylib' /usr/local/lib/libag_sk.8.dylib
+sudo install_name_tool -change 'libag_gui.dylib' '/usr/local/lib/libag_gui.dylib' /usr/local/lib/libag_sk.8.dylib
+sudo install_name_tool -change 'libag_core.dylib' '/usr/local/lib/libag_core.dylib' /usr/local/lib/libag_sk.8.dylib
+sudo install_name_tool -change 'libag_core.dylib' '/usr/local/lib/libag_core.dylib' /usr/local/lib/libag_vg.8.dylib
+sudo install_name_tool -change 'libag_gui.dylib' '/usr/local/lib/libag_gui.dylib' /usr/local/lib/libag_vg.8.dylib
-- 
2.39.3 (Apple Git-146)

